geoopt/manifolds/poincare/__init__.py

```diff
@@ -86,32 +86,26 @@ def expmap(self, x, u, *, project=True, dim=-1):
     def logmap(self, x, y, *, dim=-1):
         return math.logmap(x, y, c=self.c, dim=dim)
 
-    def transp(self, x, y, v, *more, dim=-1):
-        if not more:
-            return math.parallel_transport(x, y, v, c=self.c, dim=dim)
-        else:
-            return tuple(
-                math.parallel_transport(x, y, vec, c=self.c, dim=dim)
-                for vec in (v, *more)
-            )
+    def transp(self, x, y, v, dim=-1):
+        return math.parallel_transport(x, y, v, c=self.c, dim=dim)
 
-    def transp_follow_retr(self, x, u, v, *more, dim=-1):
+    def transp_follow_retr(self, x, u, v, dim=-1):
         y = self.retr(x, u, dim=dim)
-        return self.transp(x, y, v, *more, dim=dim)
+        return self.transp(x, y, v, dim=dim)
 
-    def transp_follow_expmap(self, x, u, v, *more, dim=-1, project=True):
+    def transp_follow_expmap(self, x, u, v, dim=-1, project=True):
```

Comment:
I cannot remove these ones and rely on the default implementation because of the additional args. Any idea how to do this? It'd work by adding *args, **kwargs to the base methods, but that would be cumbersome.

A side comment is that this design is not very nice from an OOP point of view as implementation signatures shouldn't vary in sub-classes (Liskov substitution).  Maybe make them class members?

Reply:
Dim argument is hard to design. We had a lot of pain with it in experiments, where dim may appear in different places. We also had typos forgetting to provide `c` for pure functions. Projection was also a pain. Therefore we decided to put it into the class methods, providing additional arguments. 

Default signature is compatible with base class expected behavior, so we agreed on adding more arguments. Idea we followed is "some manifolds can probably do more than others"

Reply:
The challenge with dim is, in example, matrix manifold, where you have more than one manifold dim